rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -89,17 +90,53 @@ enum MetricValue {
     },
 }
 
+trait Decodable<const SIZE: usize>: Copy {
+    const SIZE: usize = SIZE;
+
+    fn decode_bytes(bytes: [u8; SIZE]) -> Self;
+}
+
+impl Decodable<4> for u32 {
+    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
+        Self::from_le_bytes(bytes)
+    }
+}
+
+impl Decodable<8> for u64 {
+    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
+        Self::from_le_bytes(bytes)
+    }
+}
+
+impl Decodable<8> for f64 {
+    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
+        Self::from_le_bytes(bytes)
+    }
+}
+
 #[derive(Debug, Deserialize)]
 #[serde(tag = "format", rename_all = "lowercase")]
 enum EncodedSeries<T> {
     Array { data: Vec<T> },
+    Base64 { data: String },
 }
 
 impl<T> EncodedSeries<T> {
-    fn into_vec(self) -> Vec<T> {
-        match self {
+    fn into_vec<const SIZE: usize>(self) -> Result<Vec<T>, anyhow::Error>
+    where
+        T: Decodable<SIZE>,
+    {
+        let res = match self {
             EncodedSeries::Array { data } => data,
-        }
+            EncodedSeries::Base64 { data, .. } => BASE64
+                .decode(data.as_bytes())?
+                .chunks_exact(T::SIZE)
+                .map(TryInto::try_into)
+                .map(Result::unwrap)
```

Comment:
A comment on why you unwrap here might be helpful:

```suggestion
                .map(Result::unwrap) // OK to unwrap, `chunks_exact` always yields slices of the right length
```

Reply:
Added, thank you!